src/test/java/org/jvnet/hudson/test/RealJenkinsRuleTest.java

```diff
@@ -179,6 +190,57 @@ private static void _stepsDoNotRunOnHttpWorkerThread(JenkinsRule r) throws Throw
         assertNull(Stapler.getCurrentRequest());
     }
 
+
+    @Test
+    public void test500Errors() throws IOException {
+        HttpURLConnection conn = mock(HttpURLConnection.class);
+        when(conn.getResponseCode()).thenReturn(500);
+        try {
+            RealJenkinsRule.checkResult(conn);
+            fail("expected exception JenkinsStartupException");
+        }catch (RealJenkinsRule.JenkinsStartupException ae){
+
+        }
+    }
+    @Test
+    public void test503Errors() throws IOException {
+        HttpURLConnection conn = mock(HttpURLConnection.class);
+        when(conn.getResponseCode()).thenReturn(503);
+        when(conn.getErrorStream()).thenReturn(IOUtils.toInputStream("Jenkins Custom Error", "UTF-8"));
+
+        Optional<String> s = RealJenkinsRule.checkResult(conn);
+
+        assertThat(s, is(Optional.of("Jenkins Custom Error")));
+    }
+
+    @Test
+    public void test200Ok() throws IOException {
+
+        HttpURLConnection conn = mock(HttpURLConnection.class);
+        when(conn.getResponseCode()).thenReturn(200);
+        when(conn.getInputStream()).thenReturn(IOUtils.toInputStream("blah blah blah", "UTF-8"));
+
+        Optional<String> s = RealJenkinsRule.checkResult(conn);
+
+        verify(conn, times(1)).getInputStream();
+        assertThat(s, is(Optional.empty()));
+    }
+
+    /**
+     * plugins/failure.hpi
+     *  Plugin that has this:
+     *
+     *  @Initializer(after=JOB_LOADED)
+     *     public static void init() throws IOException {
+     *         throw new IOException("oops");
+     *     }
+     *
+     */
+    @Test(expected = RealJenkinsRule.JenkinsStartupException.class)
```

Comment:
Is there an expected `err` in this case? (can use `assertThrows` for more control)

Reply:
not sure if the stack suppression takes care of hiding that from us or not - but if not could be worth a check.